### PR TITLE
SAK-49839 Discussions: Disappearing calendar control in topic/forum settings

### PR DESCRIPTION
**Pull request by hornersa**
Jira: https://sakaiproject.atlassian.net/browse/SAK-49839

I attached to the jira a screencast named SAK-49839-ProposedFix.mp4 to demo the result of this proposed fix. Also, I added presentation logic to not display the "Send these Open and Close dates to Calendar" checkbox if the Calendar tool is not in the site.

### CODE REVIEW

File: msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp [hunk 1]
```diff
@@ -295,19 +294,19 @@
                    <h:inputText id="closeDate" styleClass="closeDate" value="#{ForumTool.selectedForum.closeDate}" onchange="storeCloseDateISO(event)"/>
                    <h:inputText id="closeDateISO" styleClass="closeDateISO hidden" value="#{ForumTool.selectedForum.closeDateISO}"></h:inputText>
                </h:panelGroup>
-				<p class="checkbox">
+				<h:panelGroup layout="block" styleClass="checkbox" style="display: #{ForumTool.doesSiteHaveCalendar ? '' : 'none'}">
```
Comment (kunaljaykam) [inline msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp:297]: Please test it, I am not sure if jsf styleClass supports conditions
```suggestion
				<h:panelGroup layout="block" styleClass="checkbox #{ForumTool.doesSiteHaveCalendar ? '' : 'd-none'}">
```

File: msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp [hunk 2]
```diff
@@ -295,19 +294,19 @@
...
 												 value="#{ForumTool.selectedForum.forum.sendOpenCloseToCalendar}"/>
-						<h:outputLabel for="sendOpenCloseDateToCalendar" value="#{msgs.sendOpenCloseToCalendar}" />
+						<h:outputLabel for="sendOpenCloseDateToCalendar" styleClass="p-0" value="#{msgs.sendOpenCloseToCalendar}" />
 					</h:panelGroup>
-				</p>
-               <h:panelGroup id="lockForumAfterCloseDateSpan" styleClass="indnt2 lockForumAfterCloseDateSpan" style="display: #{ForumTool.selectedForum.availabilityRestricted ? '' : 'none'}">
-                   <h:selectBooleanCheckbox id="lockForumAfterCloseDate" value="#{ForumTool.selectedForum.forum.lockedAfterClosed}"/>
-                   <h:outputLabel for="lockForumAfterCloseDate" value="#{msgs.lockForumAfterCloseDate}" />
+				</h:panelGroup>
+               <h:panelGroup layout="block" id="lockForumAfterCloseDateSpan" styleClass="indnt2 lockForumAfterCloseDateSpan" style="display: #{ForumTool.selectedForum.availabilityRestricted ? '' : 'none'}">
```
Comment (kunaljaykam) [inline msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp:307]:
```suggestion
               <h:panelGroup layout="block" id="lockForumAfterCloseDateSpan" styleClass="indnt2 lockForumAfterCloseDateSpan #{ForumTool.selectedForum.availabilityRestricted ? '' : 'd-none'}">
```

File: msgcntr/messageforums-app/src/webapp/jsp/dfReviseTopicSettingsAttach.jsp [hunk 3]
```diff
@@ -285,19 +284,19 @@
 				<h:outputLabel value="#{msgs.closeDate}: " for="closeDate"/>
 				<h:inputText id="closeDate" styleClass="closeDate" value="#{ForumTool.selectedTopic.closeDate}"/>
 			</h:panelGroup>
-			<p class="checkbox">
+			<h:panelGroup styleClass="checkbox" style="display: #{ForumTool.doesSiteHaveCalendar ? '' : 'none'}">
```
Comment (kunaljaykam) [inline msgcntr/messageforums-app/src/webapp/jsp/dfReviseTopicSettingsAttach.jsp:287]:
```suggestion
			<h:panelGroup styleClass="checkbox #{ForumTool.doesSiteHaveCalendar ? '' : 'd-none'}">
```

**Review by kunaljaykam — APPROVED** [review]

File: msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp (hunk 1 above)
Comment (ottenhoff) [reply msgcntr/messageforums-app/src/webapp/jsp/dfReviseForumSettingsAttach.jsp:297]: no, this doesn't work

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**